src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -305,7 +285,7 @@ internal static string GetRuntimeDirectoryOpt()
         /// <returns>
         /// An open <see cref="NamedPipeClientStream"/> to the server process or null on failure.
         /// </returns>
-        private static NamedPipeClientStream TryConnectToServer(
+        internal static NamedPipeClientStream TryConnectToServer(
```

Comment:
This is another method that can be replaced with async pipe connect, right?

Reply:
Yes, although again, I'm trying to keep the diff as minimal as possible. (In the start of working on this PR, I was cleaning up a *lot* of stuff like this - there is a LOT - and it was impossible to keep track of what was necessary changes and what was just cleanup, which made debugging really difficult)

I plan on coming in after this PR and cleaning a lot of stuff up (now that the functionality is there) - but I've made this use ConnectAsync for this PR anyway.

(The reason why the server one was *required* to use WaitAsync was because we were relying on the behavior of "calling `Close` makes an existing `WaitForConnection` cancel", which apparently is undefined behavior and we should not be doing that (it doesn't work on Linux))